rocky/components/modal/script.js

```diff
@@ -10,67 +21,52 @@ export function initDialogs(element) {
 }
 
 export function initDialog(modal) {
-  let input_elements = [];
+  let dialog_element = modal.querySelector("dialog");
+  if (!dialog_element) return;
 
-  modal.querySelector(".modal-trigger").addEventListener("click", (event) => {
-    // Get and clone input elements to be able to "reset" them on "cancel".
-    input_elements = modal.querySelectorAll("input, textarea");
-
-    // Store the initial value in a data attribute
-    input_elements.forEach((element) => {
-      element.defaultvalue = element.value;
+  let trigger = document.querySelector(
+    "[data-modal-id='" + dialog_element.id + "']",
+  );
+  if (!trigger) return;
+  // Check if trigger element is <a>, if not, on click,
+  // alter the URL to open the dialog using the onhaschange event.
+  if (trigger.nodeName !== "A") {
```

Comment:
Should this return early? Triggers can now be `<a>`'s without a `data-modal-id`, right? In which case you would still want to init the dialog's `"click"` and `"close"` handlers, right?

```suggestion
  if (trigger && trigger.nodeName !== "A") {
```

Reply:
My reasoning for including the check on line 33 was that an element that's not an `<a>` needs to be assigned a `click` handler to function as a trigger. Whereas an `<a>` already has that logic built in.

But you point about an `<a>` without a `data-modal-id` is valid, I suppose, so the selector for `trigger` needs to be improved

Reply:
Edit: I didn't entirely comprehend the point you were making. You're totally right that returning early disables the `click` and `close` functionality tied to their respective handlers. I will fix this in my next commit :)

Reply:
Right, I suppose `"[data-modal-id='" + dialog_element.id + "']:not(a)"` would work.

I any case, should the `initDialog` function ever return early? The `dialog_element` needs its click and close handlers, regardless of whether or not there's any kind of trigger that may or may not need a click handler, right?